US-530858: Helm chart and documentation updates for updating to elasticsearch 8.x version

hsomu opened the pull request:
Helm chart and documentation updates for updating to elasticsearch 8.x version

anilkumargedela commented on charts/backingservices/charts/srs/README.md:
```diff
@@ -15,23 +15,77 @@ The service deployment provisions runtime service pods along with a dependency o
+            <td>Pega recommends using Elasticsearch version 7.17.9.</td>
+        </tr>
+        <tr>
+            <td>Basic Authentication</td>
```
Basic auth is supported by all ES versions, do we need to explicitly callout ?

hsomu replied:
Hi @anilkumargedela, ES server 7.10.2 is not supported if the K8's version is >=1.25

kingakowalska1 commented:
> > Hi Harsha,
> > Here are my comments:
> > In the table: Replace “K8’s” with “Kubernetes” Replace “Pega recommends using” with “As a best practice, use” This note is redundant “Note: To stay current with Pega releases, use the latest available SRS image v1.28” because in the table v1.28 is the only one mentioned.
> > Line 71 Replace: “To migrate to Elasticsearch version 7.17.9 or 8.10.3 from the Elasticsearch version 7.10.2 or 7.16.3 follow below steps that applies to your deployment:” With: “To migrate to Elasticsearch version 7.17.9 or 8.10.3 from the Elasticsearch version 7.10.2 or 7.16.3, perform the following steps:”
> > For the steps, instead of headers, use a simple ordered list:
> > 
> > 1. Update the SRS Docker image version to use v1.28, which supports Elasticsearch versions 7.10.x and 7.16.x.
> > 2. To update Elasticsearch version to 7.17.9, perform the following actions:
> >    a.	Update the Elasticsearch `dependencies.version` parameter in the [requirement.yaml](../../requirements.yaml) to 7.17.3.
> >    Note: This parameter references the Elasticsearch Helm chart version and not the Elasticsearch cluster version.
> >    b.	Update the `elasticsearch.imageTag` in the Backing Services Helm chart to 7.17.9.
> > 3. To update Elasticsearch version to 8.10.3, perform the following actions:
> >    a.	Update the Elasticsearch `dependencies.version` parameter in the [requirement.yaml](../../requirements.yaml) to 8.5.1.
> >    Note: This parameter references the Elasticsearch Helm chart version and not the Elasticsearch cluster version.
> >    b.	Update the `elasticsearch.imageTag` in the Backing Services Helm chart to 8.10.3.
> > 4. Restart the SRS pods.
> > 
> > Line 83 Replace “To migrate to Elasticsearch version 7.17.9 or 8.10.3 from the Elasticsearch version 7.10.2 or 7.16.3 follow below steps that applies to your deployment:” To: “To migrate to Elasticsearch version 7.17.9 or 8.10.3 from the Elasticsearch version 7.10.2 or 7.16.3, perform the following steps:”
> > For the steps, here’s a proposed simple ordered list:
> > 
> > 1. Update the SRS Docker image version to use v1.28, which supports Elasticsearch versions 7.10.x and 7.16.x.
> > 2. To use Elasticsearch version 7.17.9, upgrade your external Elasticsearch cluster to 7.17.9 according to your organization’s best practices. For more information, see official Elasticsearch version 7.17 documentation.
> > 3. To use Elasticsearch version 8.10.3, upgrade your external Elasticsearch cluster to 8.10.3 according to your organization’s best practices. For more information, see official Elasticsearch version 8.10 documentation.
> > 4. Restart the SRS pods.
> > 
> > In our official documentation, we try to avoid linking directly to third-party documentation because maintaining the links is difficult when the third party makes changes.
> > Please let me know if the recommended changes make sense. 😊
> > Thanks! Kinga
> 
> Hi Kinga, thanks for reviewing the changes. Made the changes as per the recommendations, request to re verify them. Thanks.

Hi Harsha,

I received some feedback that the following step in lines 69 and 84 of the readme is not clear:

“Update the SRS Docker image version to use v1.29, which supports both Elasticsearch versions 7.10.x and 7.16.x.”

So I propose to change it to the following for both sets of steps.

“Update the SRS Docker image version to use v1.29. This version has backward compatibility with Elasticsearch versions 7.10.x and 7.16.x, so your SRS will continue to work even before you update your Elasticsearch service.”

Does this sound correct? 

Also, could you please confirm which is the latest version of SRS, v1.29 or v.1.28? The recently published PR [https://github.com/pegasystems/pega-helm-charts/pull/675](https://github.com/pegasystems/pega-helm-charts/pull/675%20changed%20it%20to%20v.1.28) changed it to v.1.28, is it already updated to v.1.29?

Thanks!
Kinga

ReddySrini commented:
The version of SRS that has the changes supporting these changes is SRS v1.29.1

hsomu commented:
> > > Hi Harsha,
> > > Here are my comments:
> > > In the table: Replace “K8’s” with “Kubernetes” Replace “Pega recommends using” with “As a best practice, use” This note is redundant “Note: To stay current with Pega releases, use the latest available SRS image v1.28” because in the table v1.28 is the only one mentioned.
> > > Line 71 Replace: “To migrate to Elasticsearch version 7.17.9 or 8.10.3 from the Elasticsearch version 7.10.2 or 7.16.3 follow below steps that applies to your deployment:” With: “To migrate to Elasticsearch version 7.17.9 or 8.10.3 from the Elasticsearch version 7.10.2 or 7.16.3, perform the following steps:”
> > > For the steps, instead of headers, use a simple ordered list:
> > > 
> > > 1. Update the SRS Docker image version to use v1.28, which supports Elasticsearch versions 7.10.x and 7.16.x.
> > > 2. To update Elasticsearch version to 7.17.9, perform the following actions:
> > >    a.	Update the Elasticsearch `dependencies.version` parameter in the [requirement.yaml](../../requirements.yaml) to 7.17.3.
> > >    Note: This parameter references the Elasticsearch Helm chart version and not the Elasticsearch cluster version.
> > >    b.	Update the `elasticsearch.imageTag` in the Backing Services Helm chart to 7.17.9.
> > > 3. To update Elasticsearch version to 8.10.3, perform the following actions:
> > >    a.	Update the Elasticsearch `dependencies.version` parameter in the [requirement.yaml](../../requirements.yaml) to 8.5.1.
> > >    Note: This parameter references the Elasticsearch Helm chart version and not the Elasticsearch cluster version.
> > >    b.	Update the `elasticsearch.imageTag` in the Backing Services Helm chart to 8.10.3.
> > > 4. Restart the SRS pods.
> > > 
> > > Line 83 Replace “To migrate to Elasticsearch version 7.17.9 or 8.10.3 from the Elasticsearch version 7.10.2 or 7.16.3 follow below steps that applies to your deployment:” To: “To migrate to Elasticsearch version 7.17.9 or 8.10.3 from the Elasticsearch version 7.10.2 or 7.16.3, perform the following steps:”
> > > For the steps, here’s a proposed simple ordered list:
> > > 
> > > 1. Update the SRS Docker image version to use v1.28, which supports Elasticsearch versions 7.10.x and 7.16.x.
> > > 2. To use Elasticsearch version 7.17.9, upgrade your external Elasticsearch cluster to 7.17.9 according to your organization’s best practices. For more information, see official Elasticsearch version 7.17 documentation.
> > > 3. To use Elasticsearch version 8.10.3, upgrade your external Elasticsearch cluster to 8.10.3 according to your organization’s best practices. For more information, see official Elasticsearch version 8.10 documentation.
> > > 4. Restart the SRS pods.
> > > 
> > > In our official documentation, we try to avoid linking directly to third-party documentation because maintaining the links is difficult when the third party makes changes.
> > > Please let me know if the recommended changes make sense. 😊
> > > Thanks! Kinga
> > 
> > 
> > Hi Kinga, thanks for reviewing the changes. Made the changes as per the recommendations, request to re verify them. Thanks.
> 
> Hi Harsha,
> 
> I received some feedback that the following step in lines 69 and 84 of the readme is not clear:
> 
> “Update the SRS Docker image version to use v1.29, which supports both Elasticsearch versions 7.10.x and 7.16.x.”
> 
> So I propose to change it to the following for both sets of steps.
> 
> “Update the SRS Docker image version to use v1.29. This version has backward compatibility with Elasticsearch versions 7.10.x and 7.16.x, so your SRS will continue to work even before you update your Elasticsearch service.”
> 
> Does this sound correct?
> 
> Also, could you please confirm which is the latest version of SRS, v1.29 or v.1.28? The recently published PR [https://github.com/pegasystems/pega-helm-charts/pull/675](https://github.com/pegasystems/pega-helm-charts/pull/675%20changed%20it%20to%20v.1.28) changed it to v.1.28, is it already updated to v.1.29?
> 
> Thanks! Kinga

Hi Kinga, we made the necessary changes and also the version of SRS that has the changes supporting these changes is SRS v1.29.1 and updated the same.

Thanks & Regards
Harsha Vardhan Somu

kingakowalska1 reviewed (approved):
Great work! Thank you for the clarification and making the requested changes.
